s2/loop.go

```diff
@@ -1701,16 +1702,16 @@ func hasCrossingRelation(a, b *Loop, relation loopRelation) bool {
 					return true
 				}
 			} else {
-				// The A and B cells are the same. Since the two cells
-				// have the same center point P, check whether P satisfies
-				// the crossing targets.
-				aClipped := ai.it.IndexCell().shapes[0]
-				bClipped := bi.it.IndexCell().shapes[0]
-				if containsCenterMatches(aClipped, ab.aCrossingTarget) &&
-					containsCenterMatches(bClipped, ab.bCrossingTarget) {
+				// The A and B cells are the same. Since the two
+				// cells have the same center point P, check
+				// whether P satisfies the crossing targets.
+				if containsCenterMatches(ai.containsCenter(), ab.aCrossingTarget) &&
+					containsCenterMatches(bi.containsCenter(), ab.bCrossingTarget) {
 					return true
 				}
 				// Otherwise test all the edge crossings directly.
+				aClipped := ai.it.IndexCell().shapes[0]
```

Comment:
Why not use your new ai/bi.clipped() function?

Reply:
Done